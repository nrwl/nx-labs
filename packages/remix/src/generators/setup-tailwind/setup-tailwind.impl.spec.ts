import { readJson } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import applicationGenerator from '../application/application.impl';
import setupTailwind from './setup-tailwind.impl';

describe('setup-tailwind generator', () => {
  it('should add a tailwind config to an application correctly', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace();
    await applicationGenerator(tree, {
      name: 'test',
      rootProject: true,
    });

    // ACT
    await setupTailwind(tree, { project: 'test' });

    // ASSERT
    expect(tree.exists('tailwind.config.ts')).toBeTruthy();
    expect(tree.read('tailwind.config.ts', 'utf-8')).toMatchSnapshot();
    expect(tree.exists('app/tailwind.css')).toBeTruthy();
    expect(tree.read('app/tailwind.css', 'utf-8')).toMatchSnapshot();
    expect(tree.read('app/root.tsx', 'utf-8')).toMatchSnapshot();
    expect(tree.read('remix.config.js', 'utf-8')).toMatchSnapshot();
    expect(
      readJson(tree, 'package.json').dependencies['tailwindcss']
    ).toBeTruthy();
  });

  it('should add a js tailwind config to an application correctly', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace();
    await applicationGenerator(tree, {
      name: 'test',
      js: true,
      rootProject: true,
    });

    // ACT
    await setupTailwind(tree, { project: 'test', js: true });

    // ASSERT
    expect(tree.exists('tailwind.config.js')).toBeTruthy();
    expect(tree.read('tailwind.config.js', 'utf-8')).toMatchSnapshot();
    expect(tree.exists('app/tailwind.css')).toBeTruthy();
    expect(tree.read('app/tailwind.css', 'utf-8')).toMatchSnapshot();
    expect(tree.read('app/root.js', 'utf-8')).toMatchSnapshot();
    expect(tree.read('remix.config.js', 'utf-8')).toMatchSnapshot();
    expect(
      readJson(tree, 'package.json').dependencies['tailwindcss']
    ).toBeTruthy();
  });
});
