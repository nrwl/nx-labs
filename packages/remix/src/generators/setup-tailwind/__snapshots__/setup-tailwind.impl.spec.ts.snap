// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setup-tailwind generator should add a js tailwind config to an application correctly 1`] = `
"import { createGlobPatternsForDependencies } from '@nx/react/tailwind';
export default {
  content: [
    './app/**/*.{js,jsx,ts,tsx}',
    ...createGlobPatternsForDependencies(__dirname),
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
"
`;

exports[`setup-tailwind generator should add a js tailwind config to an application correctly 2`] = `
"@tailwind base;
@tailwind components;
@tailwind utilities;
"
`;

exports[`setup-tailwind generator should add a js tailwind config to an application correctly 3`] = `
"import {
  Links,
  LiveReload,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from '@remix-run/react';
import styles from './tailwind.css';
export const links = () => [{ rel: 'stylesheet', href: styles }];
export const meta = () => ({
  charset: 'utf-8',
  title: 'New Remix App',
  viewport: 'width=device-width,initial-scale=1',
});
export default function App() {
  return (
    <html lang=\\"en\\">
      <head>
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}
"
`;

exports[`setup-tailwind generator should add a js tailwind config to an application correctly 4`] = `
"/**
 * @type {import('@remix-run/dev').AppConfig}
 */
module.exports = {
  tailwind: true,
  ignoredRouteFiles: ['**/.*'],
  // appDirectory: \\"app\\",
  // assetsBuildDirectory: \\"public/build\\",
  // serverBuildPath: \\"build/index.js\\",
  // publicPath: \\"/build/\\",
  watchPaths: () => require('@nx/remix').createWatchPaths(__dirname),
};
"
`;

exports[`setup-tailwind generator should add a tailwind config to an application correctly 1`] = `
"import type { Config } from \\"tailwindcss\\";
import { createGlobPatternsForDependencies } from '@nx/react/tailwind';

export default {
  content: [
    \\"./app/**/*.{js,jsx,ts,tsx}\\",
    ...createGlobPatternsForDependencies(__dirname)
  ],
  theme: {
    extend: {},
  },
  plugins: [],
} satisfies Config;
"
`;

exports[`setup-tailwind generator should add a tailwind config to an application correctly 2`] = `
"@tailwind base;
@tailwind components;
@tailwind utilities;
"
`;

exports[`setup-tailwind generator should add a tailwind config to an application correctly 3`] = `
"import type { LinksFunction, MetaFunction } from '@remix-run/node';
import {
  Links,
  LiveReload,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from '@remix-run/react';
import styles from './tailwind.css';
export const links: LinksFunction = () => [{ rel: 'stylesheet', href: styles }];

export const meta: MetaFunction = () => ({
  charset: 'utf-8',
  title: 'New Remix App',
  viewport: 'width=device-width,initial-scale=1',
});

export default function App() {
  return (
    <html lang=\\"en\\">
      <head>
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}
"
`;

exports[`setup-tailwind generator should add a tailwind config to an application correctly 4`] = `
"/**
 * @type {import('@remix-run/dev').AppConfig}
 */
module.exports = {
  tailwind: true,
  ignoredRouteFiles: ['**/.*'],
  // appDirectory: \\"app\\",
  // assetsBuildDirectory: \\"public/build\\",
  // serverBuildPath: \\"build/index.js\\",
  // publicPath: \\"/build/\\",
  watchPaths: () => require('@nx/remix').createWatchPaths(__dirname),
};
"
`;
